torch/csrc/jit/codegen/cuda/lower_loops.h

```diff
@@ -44,6 +44,20 @@ class TORCH_CUDA_API LoopNestGenerator : public OptOutDispatch {
   // initialization
   ThreadPredicateMap& thread_predicates_;
 
+  // Fusion shared_memory values
+  // Tracks if shared memory is modified
+  std::unordered_map<Val*, bool> smem_;
```

Comment:
Looks like what you need is just a set of `Val*`s. Do you really need to have mapping to bool?

Reply:
I thought the bool was checking if it was modified but hasn't been synced.

Reply:
Yes, the bool is for checking if the buffer is modified but not synced.

Reply:
The mapped bool value is initialized to `false` first, and then is modified to `true` when modified. But, looks like it is never modified back to false, as far as I see. There is a function `cleanSharedMemory`, but I don't see it's used anywhere.